src/Nethermind/Nethermind.Trie/NibbleExtensions.cs

```diff
@@ -84,5 +84,33 @@ public static byte ToByte(Nibble highNibble, Nibble lowNibble)
         {
             return (byte)(((byte)highNibble << 4) | (byte)lowNibble);
         }
+
+        public static byte[] ToBytes(byte[] nibbles)
+        {
+            byte[] bytes = new byte[nibbles.Length / 2];
+            for (int i = 0; i < bytes.Length; i++)
+            {
+                bytes[i] = ToByte(nibbles[2 * i], nibbles[2 * i + 1]);
+            }
+
+            return bytes;
```

Comment:
use i+=2 instead of 2*i, will be faster

Reply:
Interesting, could you elaborate?

Reply:
If we add another counter j+=2, we will gain just 1% of speed up.